From 5d0f379161363f8f9a5aad7701b9140b64af414a Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Sun, 27 Nov 2011 15:42:07 +0000
Subject: Skip interval timer tests in Time::HiRes on GNU/Hurd

GNU/Hurd has no support for virtual interval timers (ITIMER_VIRTUAL)
so skip the tests on this platform along with QNX and Haiku.

Bug-Debian: http://bugs.debian.org/650094
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=72754

Patch-Name: fixes/hurd_skip_itimer_virtual.diff
---
 cpan/Time-HiRes/HiRes.pm  |    4 ++--
 cpan/Time-HiRes/t/HiRes.t |    1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/cpan/Time-HiRes/HiRes.pm b/cpan/Time-HiRes/HiRes.pm
index 8f2ec21..937e202 100644
--- a/cpan/Time-HiRes/HiRes.pm
+++ b/cpan/Time-HiRes/HiRes.pm
@@ -560,8 +560,8 @@ seconds. Time::HiRes will notice this eventually and recalibrate.
 Note that since Time::HiRes 1.77 the clock_gettime(CLOCK_MONOTONIC)
 might help in this (in case your system supports CLOCK_MONOTONIC).
 
-Some systems have APIs but not implementations: for example QNX and Haiku
-have the interval timer APIs but not the functionality.
+Some systems have APIs but not implementations: for example QNX, Haiku
+and GNU/Hurd have the interval timer APIs but not the functionality.
 
 =head1 SEE ALSO
 
diff --git a/cpan/Time-HiRes/t/HiRes.t b/cpan/Time-HiRes/t/HiRes.t
index 0a38e78..cfd05fc 100644
--- a/cpan/Time-HiRes/t/HiRes.t
+++ b/cpan/Time-HiRes/t/HiRes.t
@@ -353,6 +353,7 @@ unless (defined &Time::HiRes::setitimer
 	&& $Config{sig_name} =~ m/\bVTALRM\b/
 	&& $^O ne 'nto' # nto: QNX 6 has the API but no implementation
 	&& $^O ne 'haiku' # haiku: has the API but no implementation
+	&& $^O ne 'gnu' # GNU/Hurd: has the API but no implementation; <http://savannah.gnu.org/task/?8916>
     ) {
     for (18..19) {
 	print "ok $_ # Skip: no virtual interval timers\n";
